src/browser/app-state.js

```diff
@@ -20,6 +20,9 @@ var onSetConnectionState = function(currentState, connectionState){
 };
 
 var onAddTeam = function(currentState, team){
+  if (!team.name) {
+    team.name = team.url.replace(/[^:]*:\/\/[^\/]*\/([^/]*)(\/.*|$)/, '$1')
```

Comment:
> This check should be inside the src/load-settings.js.

This is where I started at first but I found it suboptimal to synthesise `name` here, plus I was in need to push out a new backwards-compatible build to my users ASAP.